### ISSUE TITLE
Cannot deploy metadata. Reason: Failed to save modifications to the server. Error returned: 'The ImpersonationInfo for datasource contains an ImpersonationMode that can only be used by a server administrator.

### ISSUE THREAD

**Issue by gflores1023** [otykier/TabularEditor#630]
**Is your feature request related to a problem? Please describe.**
When deploying a model to Azure AS (maybe on-prem), a user who is not a server admin receives this error message below. 

`Cannot deploy metadata. Reason: Failed to save modifications to the server. Error returned: 'The ImpersonationInfo for datasource contains an ImpersonationMode that can only be used by a server administrator.`

I've only seen this happen using the new Power Query connections, not the legacy connections.  The user is an admin on the database but not on the server. We've had to switch the connection to legacy so we can deploy successfully.

**Describe the solution you'd like**
[This link](https://selfservedbi.com/2018/12/22/workarounds-for-analysis-services-database-admins-without-server-admin-access/) from 2018 mentions this issue and a possible workaround with BISM-Normalizer. I've not tried this yet, though. There is also comment there that mentions another possible workaround ("impersonateUnattended").

**Additional context**
I don't know if this is a bug in SSAS or if this is fix-able by Tabular Editor. We can't use the new Power Query connections due to this issue. (Fortunately, we haven't had a need).

**Comment by otykier** [comment]
Tabular Editor generates a TMSL script for deployment purposes. You can export the TMSL script in order to try to manually execute it from SSMS. If you get the same error when manually executing through SSMS, then this is indeed a limitation of the product, as Tabular Editor does not have other ways of performing a deployment (I am not interested in bringing in ADOMDClient or other libraries that ALM Toolkit may be using to work around this limitation).

Alternatively, if setting the impersonationMode to "impersonateUnattended" is a viable solution, then why not set that value, before deploying?

**Comment by gflores1023** [comment]
Correct, we got the error when running in SSMS. I actually just discovered this article and possible workaround when I posted this. We made our model connection changes a couple  weeks ago so we haven't messed with it to try this. When I get some free time, I'll try those things.

In any case, I wanted to make a post for visibility in case other people ran into this.
